test/src/unit-cppapi-deletes.cc

```diff
@@ -1455,4 +1503,40 @@ TEST_CASE_METHOD(
   CHECK(!memcmp(c_dim2.data(), dim2.data(), c_dim2.size() * sizeof(uint64_t)));
 
   remove_sparse_array();
-}
\ No newline at end of file
+}
+
+#ifndef _WIN32
+TEST_CASE_METHOD(
+    DeletesFx,
+    "CPP API: Test writing delete in middle of fragment consolidated without "
+    "timestamps",
+    "[cppapi][deletes][write][old-consolidated-fragment]") {
+  if constexpr (is_experimental_build) {
+    return;
+  }
+
+  remove_sparse_array();
+  create_sparse_array_v11();
+  // Write first fragment.
+  write_sparse_v11(1);
+
+  // Write second fragment.
+  write_sparse_v11(3);
+
+  // Consolidate.
+  consolidate_sparse();
+
+  // Upgrade to latest version.
+  Array::upgrade_version(ctx_, SPARSE_ARRAY_NAME);
+
+  // Define query condition (d2 == 3).
+  QueryCondition qc2(ctx_);
+  uint64_t val2 = 3;
+  qc2.init("d2", &val2, sizeof(uint64_t), TILEDB_EQ);
+
+  // Write condition and specify an error is expected.
```

Comment:
nit: I would suggest to remove the `error_expected` argument to this helper function, and handle both cases with `REQUIRE_THROWS{_AS}` in the test body itself.